server_fn/src/codec/msgpack.rs

```diff
@@ -0,0 +1,74 @@
+use super::{Encoding, FromReq, FromRes, IntoReq, IntoRes};
+use crate::{
+    error::ServerFnError,
+    request::{ClientReq, Req},
+    response::{ClientRes, Res},
+};
+use bytes::Bytes;
+use http::Method;
+use serde::{de::DeserializeOwned, Serialize};
+
+/// A codec for MessagePack.
+pub struct MsgPack;
+
+impl Encoding for MsgPack {
+    const CONTENT_TYPE: &'static str = "application/msgpack";
+    const METHOD: Method = Method::POST;
+}
+
+impl<T, Request, Err> IntoReq<MsgPack, Request, Err> for T
+where
+    Request: ClientReq<Err>,
+    T: Serialize,
+{
+    fn into_req(
+        self,
+        path: &str,
+        accepts: &str,
+    ) -> Result<Request, ServerFnError<Err>> {
+        let data = rmp_serde::to_vec(&self)
+            .map_err(|e| ServerFnError::Serialization(e.to_string()))?;
+        Request::try_new_post_bytes(
+            path,
+            MsgPack::CONTENT_TYPE,
+            accepts,
+            Bytes::from(data),
+        )
+    }
+}
+
+impl<T, Request, Err> FromReq<MsgPack, Request, Err> for T
+where
+    Request: Req<Err> + Send,
+    T: DeserializeOwned,
+{
+    async fn from_req(req: Request) -> Result<Self, ServerFnError<Err>> {
+        let string_data = req.try_into_string().await?;
```

Comment:
Shouldn't this use [`try_into_bytes`](https://docs.rs/server_fn/latest/server_fn/request/trait.Req.html#tymethod.try_into_bytes) instead? I would think `try_into_string` fails if the MessagePack data is not valid UTF-8. I don't know the format but that seems unlikely.

Reply:
You were correct. It is not valid UTF-8.